### ISSUE TITLE
Sorting does not seem to be considered when use `DISTINCT ON`

### ISSUE THREAD

**Issue by eitsupi** [PRQL/prql#2711]
### What happened?

From https://github.com/PRQL/prql/pull/2708#discussion_r1215611454

I used the latest dev version (5886a758b66422a1a398dcbad7c7deaa43022b90).

### PRQL input

```elm
prql target:sql.duckdb

from employees
group department (
  sort age
  take 1
)
```

### SQL output

```SQL
SELECT
  DISTINCT ON (department) *
FROM
  employees
```

### Expected SQL output

```SQL
SELECT
  DISTINCT ON (department) *
FROM
  employees
ORDER BY
  department,
  age
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by aljazerzen** [comment]
Was distinct on even tested in a database?

**Comment by max-sixty** [comment]
This is my fault! I'm looking now

**Comment by max-sixty** [comment]
Closed by #2715

**Comment by eitsupi** [comment]
Thanks!
